src/libstd/process.rs

```diff
@@ -0,0 +1,835 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! Working with processes.
+
+#![unstable(feature = "process", reason = "recently added via RFC 579")]
+#![allow(non_upper_case_globals)]
+
+use core::prelude::*;
+
+use env;
+use ffi::AsOsStr;
+use fmt;
+use old_io::{self, Reader, Writer};
+use io::{Error, ErrorKind};
+use io::Result as IoResult;
+use path::AsPath;
+use libc;
+use sync::mpsc::{channel, Receiver};
+use sys::fs::FileDesc;
+use sys::process2::Process as ProcessImp;
+use sys::process2::Command as CommandImp;
+use sys;
+use sys_common::{AsInner, AsInnerMut};
+use thread::Thread;
+use vec::Vec;
+
+/// Signal a process to exit immediately, forcibly killing it. Corresponds to
+/// SIGKILL on unix platforms.
+#[cfg(windows)] const MustDieSignal: int = 9;
+/// Signal a process to exit immediately, forcibly killing it. Corresponds to
+/// SIGKILL on unix platforms.
+#[cfg(not(windows))] const MustDieSignal: int = libc::SIGKILL as int;
```

Comment:
I think that in general we've had pretty good success with trying to have as few `#[cfg]` directives as possible, could this logic be pushed into each respective `sys` module?